src/libstd/io/cursor.rs

```diff
@@ -15,16 +15,63 @@ use cmp;
 use io::{self, SeekFrom, Error, ErrorKind};
 use slice;
 
-/// A `Cursor` is a type which wraps a non-I/O object to provide a `Seek`
-/// implementation.
+/// A `Cursor` wraps another type and provides it with a [`Seek`][seek]
+/// implementation. This is often done to allow I/O operations on a non-I/O
+/// type.
```

Comment:
This may want to be rephrased a bit because the primitive types (`Vec<u8>`, `&[u8]`, etc) already implement I/O traits. The main functionality added by `Cursor` is the ability to seek.